WordPress/src/main/java/org/wordpress/android/networking/menus/MenusRestWPCom.java

```diff
@@ -0,0 +1,162 @@
+                mDelegate.onMenuCreated(HttpURLConnection.HTTP_OK, menu);
+                // TODO: call updateMenu if the menu has any non-default fields
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override public void onErrorResponse(VolleyError error) {
+                mDelegate.onMenuCreated(statusCodeFromVolleyError(error), menu);
+            }
+        });
+        return true;
+    }
+
+    public boolean updateMenu(final MenuModel menu) {
+        if (menu == null) return false;
+        String path = formatPath(MENU_REST_PATH, String.valueOf(menu.menuId));
+        Map<String, String> params = new HashMap<>();
+        params.put(MENU_NAME_KEY, menu.name);
+        WordPress.getRestClientUtilsV1_1().post(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                MenuModel menu = menuFromJson(response.optJSONObject(MENU_KEY));
+                mDelegate.onMenuUpdated(HttpURLConnection.HTTP_OK, menu);
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override public void onErrorResponse(VolleyError error) {
+                mDelegate.onMenuUpdated(statusCodeFromVolleyError(error), menu);
+            }
+        });
+        return true;
+    }
+
+    public void fetchMenu(long menuId) {
+        String path = formatPath(MENU_REST_PATH, String.valueOf(menuId));
+        Map<String, String> params = new HashMap<>();
+        WordPress.getRestClientUtilsV1_1().get(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                JSONObject menuObject = response.optJSONObject(MENU_KEY);
+                mDelegate.onMenuReceived(HttpURLConnection.HTTP_OK, menuFromJson(menuObject));
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override public void onErrorResponse(VolleyError error) {
+                mDelegate.onMenuReceived(statusCodeFromVolleyError(error), null);
+            }
+        });
+    }
+
+    public void fetchAllMenus() {
+        String path = formatPath(MENUS_REST_PATH, null);
+        Map<String, String> params = new HashMap<>();
+        WordPress.getRestClientUtilsV1_1().get(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                JSONArray menusJson = response.optJSONArray(ALL_MENUS_MENUS_KEY);
+                List<MenuModel> menus = new ArrayList<>();
+                try {
+                    for (int i = 0; i < menusJson.length(); ++i) {
+                        menus.add(menuFromJson(menusJson.getJSONObject(i)));
+                    }
+                } catch (JSONException exception) {
+                    AppLog.w(AppLog.T.API, "Error parsing All Menus REST response");
+                }
+                mDelegate.onMenusReceived(HttpURLConnection.HTTP_OK, menus);
```

Comment:
Idea for improvement/discussion: could the delegate be simplified by generalising and only having `onMenusReceived`, which in the specific case of fetching one element would return an `ArrayList`containting  `1` element  instead of having two different delegates? (I'm referring to `onMenuReceived` and `onMenusRecevied`). I'm generally good either way, but I'd like to check what you think in terms of separation vs. easiness of implementation. I don't think we are forced to map the WP REST API entirely, we _could_ make the consumer of this class's life easier if we want.